### ISSUE TITLE
After published 2.7.1  old versions have stopped being build

### ISSUE THREAD

**Issue by DmitriyBetskikh** [mParticle/react-native-mparticle#71]
...node_modules/react-native-mparticle/android/src/main/java/com/mparticle/react/MParticleModule.java:499: error: cannot find symbol
                builder.info(customInfo);

**Comment by DmitriyBetskikh** [comment]
<img width="790" alt="image" src="https://user-images.githubusercontent.com/102299488/169275739-3db4d82c-5f9e-4f94-9e0e-9a8eb8a8f11e.png">
This changes are breaking the build

**Comment by DmitriyBetskikh** [comment]
I am using "react-native-mparticle" ver 2.4.12. Yesterday the build in the AppCenter was successful, and today the same code fails. Those changes broken v. 2.4.12. I created a patch, revert build.gradle changes and it solved my problem.

**Comment by MichalCiesliczka** [comment]
@ngregory-rbi your change was reverted in here
https://github.com/mParticle/react-native-mparticle/commit/62ccd46adc2235518e7cdf0679af54fcca4489ff

**Comment by ngregory-rbi** [comment]
> @ngregory-rbi your change was reverted in here [62ccd46](https://github.com/mParticle/react-native-mparticle/commit/62ccd46adc2235518e7cdf0679af54fcca4489ff)

@MichalCiesliczka I see that now. Regardless, the point still stands that the fix I pointed to was intended to fix the error at the root of this issue. My change was only reverted when the actual source of the error was resolved as well (in that same commit). There's a lot of moving parts and it appears Dmitriy is pointing to the wrong ones. It's a moot issue anyway now. Pinging @willpassidomo to close this issue (unless there needs to be further investigation).

**Comment by willpassidomo** [comment]
Hey @DmitriyBetskikh just to clarify, you are using `react-native-mparticle` version 2.7.1 as well as MParticle's `android-core` version 5.39.0? We removed some a few deprecated methods in that Android version, so compatibility is broken between `react-nave-mparticle` < 2.7.1 and `android-core` >= 5.39.0

@ngregory-rbi I believe you are correct, I fast-followed your PR with one of my own that cleaned up the problem method so the upper bound on the transitive dependency is no longer necessary. I'm realizing now that I should update the **lower bound** based on my previous paragraph. Also, I missed it in my original review, but that transitive dependency likely isn't gating which `android-core` version is actually pulled in, just due to the fact it's a compile-time dependency, whichever version the `implementation`/`api` directive in your project is pointing towards is almost definitely going to be the one that is consumed by the project so I think our setting there is likely moot

**Comment by MichalCiesliczka** [comment]
I'm using `react-nave-mparticle@2.4.12` and `com.mparticle:android-core:5.16.5` and still getting that error.
Based on what @willpassidomo said in the 2nd paragraph the 2nd dependency version is not important, because my project can still be using the version that deprecates `. info` method?

**Comment by willpassidomo** [comment]
If anyone is stumbling upon this issue with a similar problem, just to reiterate, the solution wouldd be to either:

1) update `react-native-mparticle` to the latest version (`2.7.1`) AND update the Android Gradle `com.mparticle:android-core` dependency to the latest version (`5.39.0`)

OR, if you are unable to update `react-native-mparticle`

- Hardcoed the `com.mparticle:android-core` dependency to version `5.38.2` or earlier. To check which version of this library your application is ultimatley consuming, run `./gradlew dependencies` (or `./gradlew :{module}:dependencies`).

If you are still having problems ensuring `com.mparticle:android-core` is following your version directive, you can force it by adding:

```
configurations.all {
    resolutionStrategy {
        force("com.mparticle:android-core:5.38.2")
    }
}
    ```
    to the top-level of your root or module's `build.gradle` file (sample is`kts` compatible)

**Comment by willpassidomo** [comment]
update: rolling back Android SDK change in PR: https://github.com/mParticle/mparticle-android-sdk/pull/144/files

**Comment by willpassidomo** [comment]
Rollback was released in version 5.40.1`. This version contains the `info()`/`setInfo()`/`getInfo()` methods, so it is be compatible with all versions of `react-native-mparticle`. This just went out, so expect it to be available on Maven Central within the next hour or so

If anyone here is still experiencing issues, please ping me here and I will reopen the thread. Thanks for all the feedback!
